### PR TITLE
ci: basic semgrep configuration

### PR DESCRIPTION
**Pull request by mr-uniswap**
## Description
Testing out Semgrep on interface.  This adds a semgrep github action.

### CODE REVIEW

File: .github/workflows/semgrep.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Semgrep
+on:
+  workflow_dispatch: {}
+  pull_request: {}
+  push:
+    branches:
+      - main
+      - master
```
Comment (cartcrom) [inline .github/workflows/semgrep.yml:8]: prob don't need master here?

**Review by cartcrom — APPROVED** [review]

File: .github/workflows/semgrep.yml (hunk 1 above)
Comment (mr-uniswap) [reply .github/workflows/semgrep.yml:8]: I see, I'll remove that.

File: .github/workflows/semgrep.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Semgrep
+on:
+  workflow_dispatch: {}
+  pull_request: {}
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/semgrep.yml
```
Comment (kristiehuang) [inline .github/workflows/semgrep.yml:9]: is `paths` used correctly here? https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

Comment (mr-uniswap) [reply .github/workflows/semgrep.yml:9]: Yeah, this is correct.

**Review by kristiehuang — APPROVED** [review]

**Pull request merged**